zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1027,6 +1029,10 @@ private void validateCreateRequest(String path, CreateMode createMode, Request r
                         rv.add(new ACL(a.getPerms(), cid));
                     }
                 }
+                if (Arrays.stream(ZNodeGroupAclUtil.getOpenReadAccessPathPrefixes()).anyMatch(path::contains)) {
```

Comment:
Let's add some descriptive comments here. Also should we consider making this entirely optional?

Reply:
I already made this optional by return an empty array when reading the config so we don't need to explicitly mark it here. When this property is not configured, this line will be like `Arrays.stream(new` `String[0]).anyMatch(path::startWith).`